cmake/modules/FindGMock.cmake

```diff
@@ -113,172 +117,329 @@ function(GTEST_ADD_TESTS executable extra_args)
+            string(REGEX MATCH "${gtest_test_type_regex}" test_type ${hit})
+
+            # Parameterized tests have a different signature for the filter
+            if("x${test_type}" STREQUAL "xTEST_P")
+                string(REGEX REPLACE ${gtest_case_name_regex} "*/\\1.\\2/*" test_name ${hit})
+            elseif("x${test_type}" STREQUAL "xTEST_F" OR "x${test_type}" STREQUAL "xTEST")
+                string(REGEX REPLACE ${gtest_case_name_regex} "\\1.\\2" test_name ${hit})
+            elseif("x${test_type}" STREQUAL "xTYPED_TEST")
+                string(REGEX REPLACE ${gtest_case_name_regex} "\\1/*.\\2" test_name ${hit})
+            else()
+                message(WARNING "Could not parse GTest ${hit} for adding to CTest.")
+                continue()
+            endif()
+            add_test(NAME ${test_name} COMMAND ${executable} --gtest_filter=${test_name} ${extra_args})
         endforeach()
     endforeach()
 endfunction()
 
-include(ExternalProject)
+function(_append_debugs _endvar _library)
+    if(${_library} AND ${_library}_DEBUG)
+        set(_output optimized ${${_library}} debug ${${_library}_DEBUG})
+    else()
+        set(_output ${${_library}})
+    endif()
+    set(${_endvar} ${_output} PARENT_SCOPE)
+endfunction()
 
-if(UNIX)
-    find_package(Threads REQUIRED) # GTest needs this and it's a static library
-    set(Suffix ".a")
-else()
-    set(Suffix ".lib")
+function(_gmock_find_library _name)
+    find_library(${_name}
+        NAMES ${ARGN}
+        HINTS
+        ENV GMOCK_ROOT
+        ${GMOCK_ROOT}
+        PATH_SUFFIXES ${_gmock_libpath_suffixes}
+        )
+    mark_as_advanced(${_name})
+endfunction()
+
+function(_gtest_find_library _name)
+    find_library(${_name}
+        NAMES ${ARGN}
+        HINTS
+        ENV GTEST_ROOT
+        ${GTEST_ROOT}
+        PATH_SUFFIXES ${_gtest_libpath_suffixes}
+        )
+    mark_as_advanced(${_name})
+endfunction()
+
+if(NOT DEFINED GMOCK_MSVC_SEARCH)
+    set(GMOCK_MSVC_SEARCH MD)
 endif()
 
-if(GTEST_USE_STATIC_LIBS)
-  set(GTEST_CMAKE_ARGS -Dgtest_force_shared_crt:BOOL=ON -DBUILD_SHARED_LIBS=OFF)
+set(_gmock_libpath_suffixes lib)
+set(_gtest_libpath_suffixes lib)
+if(MSVC)
+    if(GMOCK_MSVC_SEARCH STREQUAL "MD")
+        list(APPEND _gmock_libpath_suffixes
+            msvc/gmock-md/Debug
+            msvc/gmock-md/Release)
+        list(APPEND _gtest_libpath_suffixes
+            msvc/gtest-md/Debug
+            msvc/gtest-md/Release)
+    elseif(GMOCK_MSVC_SEARCH STREQUAL "MT")
+        list(APPEND _gmock_libpath_suffixes
+            msvc/gmock/Debug
+            msvc/gmock/Release)
+        list(APPEND _gtest_libpath_suffixes
+            msvc/gtest/Debug
+            msvc/gtest/Release)
+    endif()
+endif()
+
+find_path(GMOCK_INCLUDE_DIR gmock/gmock.h
+    HINTS
+    $ENV{GMOCK_ROOT}/include
+    ${GMOCK_ROOT}/include
+    )
+mark_as_advanced(GMOCK_INCLUDE_DIR)
+
+find_path(GTEST_INCLUDE_DIR gtest/gtest.h
```

Comment:
Usually in a project where cucumber-cpp is used, also a unitttest framework is used, often this might be googletest/googlemock. Therefore my intention was to be able to use this existing (already compiled) version of googlemock and googletest and not compiling it again. I have to confess, that I din't try to use installed versions, i.e. debian packages. I will check that. Thanks for the hint.